src/ATen/native/xpu/Activation.cpp

```diff
@@ -396,4 +397,48 @@ Tensor XPUNativeFunctions::hardswish_backward(
   return iter.output();
 }
 
+static void check_even_dimension(const Tensor& input, int64_t dim) {
+  auto wrap_dim = maybe_wrap_dim(dim, input.dim());
+  const int64_t nln = input.size(wrap_dim);
+  TORCH_CHECK(
+      nln % 2 == 0,
+      "Halving dimension must be even, but dimension",
+      wrap_dim,
+      " is size ",
+      nln);
+}
+
+Tensor& XPUNativeFunctions::glu_out(
+    const Tensor& self,
+    int64_t dim,
+    Tensor& out) {
+  check_even_dimension(self, dim);
```

Comment:
Should align with stock PyTorch,
<img width="596" alt="image" src="https://github.com/intel/torch-xpu-ops/assets/28250760/fbcdbde1-d5ba-46ee-b47b-ec3fc3feecc0">